src/python/WMComponent/AgentStatusWatcher/AgentStatusPoller.py

```diff
@@ -235,3 +240,34 @@ def collectWMBSInfo(self):
         logging.debug("List of jobs pending for each site, sorted by priority: %s", results['sitePendCountByPrio'])
 
         return results
+
+    def collectProxyInfo(self, agentInfo):
```

Comment:
BTW, how about we get the dates (or timeleft) AND only if any of them are going to expire, we then fetch the remaining info (like subject), and update `agentInfo`